amethyst_ui/src/text_editing.rs

```diff
@@ -62,6 +63,19 @@ impl<'a> System<'a> for TextEditingInputSystem {
                     .join()
                     .next()
             {
+                match self.last_selected {
+                    Some(last) => {
+                        if last != entity {
+                            edit_events.single_write(UiEvent::new(UiEventType::Blur, last));
```

Comment:
Focus events should be emitted from the SelectionSystem. The TextEditing system isn't responsible for focus event handling emission.